### PR TITLE
Twitter integration

### PR DESCRIPTION
**Pull request by hagane**
Got some serious progress. Figured somebody better confirm that this code is sane before I delve into thread pools and twitter's JSON.

### CODE REVIEW

File: src/main/scala/ru/org/codingteam/horta/events/TwitterEndpoint.scala [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  val client = new ClientBuilder()
+    .authentication(auth)
+    .hosts(Constants.STREAM_HOST)
+    .endpoint(apiEndpoint)
+    .processor(new StringDelimitedProcessor(queue))
+    .build()
+
+  override def start(): Unit = {
+    apiEndpoint.stallWarnings(false)
+    apiEndpoint.withFollowings(true)
```
Comment (ForNeVeR) [inline src/main/scala/ru/org/codingteam/horta/events/TwitterEndpoint.scala]: I would prefer this to be either `false` or configurable. We don't care about who have followed [@tsoding](https://twitter.com/tsoding), we care only about what this account tweets.

**Comment by ForNeVeR** [comment]
Well, currently I'd say that it's okay in principle. You can keep that way of implementation.

File: src/main/scala/ru/org/codingteam/horta/events/TwitterEndpoint.scala (hunk 1 above)
Comment (hagane) [inline src/main/scala/ru/org/codingteam/horta/events/TwitterEndpoint.scala]: It's actually meant to include tweets from everyone, who is followed by the horta, as per
https://dev.twitter.com/streaming/overview/request-parameters#with
Actually... I think we need to discuss what twitter integration means exactly =)

**Comment by hagane** [comment]
All right, that should do it. Sadly, I have no idea how to automate twitter endpoint testing.

File: src/main/scala/ru/org/codingteam/horta/events/TwitterEndpoint.scala [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      } map {
+        _.toString()
+      }
+      val tweet = message.fields.get("text") map {
+        _.toString()
+      }
+
+      (author, tweet) match {
+        case (Some(authorStr:String), Some(tweetStr:String)) => eventCollector.onEvent(TwitterEvent(authorStr, tweetStr))
+        case _ => //Probably not a tweet
```
Comment (ForNeVeR) [inline src/main/scala/ru/org/codingteam/horta/events/TwitterEndpoint.scala]: I think it would be nice to add logging here.

**Comment by ForNeVeR** [comment]
You should add a room configuration key to enable/disable Twitter plugin at least.

Ideally it would be a configuration for every room to connect to a chosen Twitter account, but currently we don't need that. Although, we already have Horta installed to multiple rooms, and only one of them need this Twitter support.

File: src/main/scala/ru/org/codingteam/horta/Application.scala [hunk 3]
```diff
@@ -20,12 +20,17 @@ import ru.org.codingteam.horta.plugins.markov.MarkovPlugin
...
-import ru.org.codingteam.horta.plugins.{VersionPlugin, AccessPlugin, FortunePlugin}
+import ru.org.codingteam.horta.plugins.{AccessPlugin, FortunePlugin, VersionPlugin}
 import ru.org.codingteam.horta.protocol.jabber.JabberProtocol
 import scalikejdbc.GlobalSettings
 
 object Application extends App with StrictLogging {
 
+  //TODO: This should be eventually handled by some kind of dependency injeciton
+  val eventEndpoints = List(
+    new TwitterEndpoint()
```
Comment (ForNeVeR) [inline src/main/scala/ru/org/codingteam/horta/Application.scala]: You should also disable `TwitterEndpoint` (or, better, even don't add it here) if Twitter isn't configured.

**Comment by hagane** [comment]
You're right. I'll see what I can do about all that.

**Comment by hagane** [comment]
All right, i've got it done. I'd like to add some tests for all this first and perform some acceptance tests myself.

Sadly, horta does not starts for me anymore failing with `Configuration not found at 'None'.`
Looks like it has no idea where to look for it's configuration and the default of `horta.properties` is not so default anymore. Can anyone verify that? Maybe it's just my working copy of the code got mucked with whatever.

**Comment by hagane** [comment]
Apparently, I'm daft. TwitterEndpoint was initialized before the configuration. We seriously have to have something to prevent these issues in the future.

After my caffeine high wears off and I will become lucid again, I'll add some test cases for these improvements.

**Comment by hagane** [comment]
I've reviewed the code. Required feature set is already covered by tests. I must gave been to high on stims back then. I'll do some manual testing next evening and report back.

**Comment by hagane** [comment]
There, I'm satisfied. Are you satisfied, Mr. @ForNeVeR ?

**Comment by ForNeVeR** [comment]
Well, let's try that!

**Pull request merged**